x-pack/plugins/cross_cluster_replication/public/app/sections/follower_index_edit/follower_index_edit.js

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { PureComponent, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { injectI18n, FormattedMessage } from '@kbn/i18n/react';
+import chrome from 'ui/chrome';
+import { MANAGEMENT_BREADCRUMB } from 'ui/management';
+
+import {
+  EuiPageContent,
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiOverlayMask,
+  EuiConfirmModal,
+} from '@elastic/eui';
+
+import { listBreadcrumb, editBreadcrumb } from '../../services/breadcrumbs';
+import routing from '../../services/routing';
+import {
+  FollowerIndexForm,
+  FollowerIndexPageTitle,
+  SectionLoading,
+  SectionError,
+  RemoteClustersProvider,
+} from '../../components';
+import { API_STATUS } from '../../constants';
+
+export const FollowerIndexEdit = injectI18n(
+  class extends PureComponent {
+    static propTypes = {
+      getFollowerIndex: PropTypes.func.isRequired,
+      selectFollowerIndex: PropTypes.func.isRequired,
+      saveFollowerIndex: PropTypes.func.isRequired,
+      clearApiError: PropTypes.func.isRequired,
+      apiError: PropTypes.object.isRequired,
+      apiStatus: PropTypes.object.isRequired,
+      followerIndex: PropTypes.object,
+      followerIndexId: PropTypes.string,
+    }
+
+    static getDerivedStateFromProps({ followerIndexId }, { lastFollowerIndexId }) {
+      if (lastFollowerIndexId !== followerIndexId) {
```

Comment:
What's this used for? If you user updates the URL to be a different ID? If so, could we add a comment to explain the intention?

Reply:
I believe this is to fetch follower index data if we don't already have it in the store, i.e. during a page reload on the edit screen. I've updated the comment here and also in the equivalent method of auto follow pattern edit

Reply:
Yes this is when we directly access the page without going first through the list of followers